### ISSUE TITLE
Illegal usage of top-level annotations in sub-level models/packages of MSL

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#191]
**Reported by dietmarw on 15 Jul 2009 08:12 UTC**
The MSL is currently flooded with the illegal use of the **top-level** annotations:
 * `version`
 * `versionDate`
 * `uses`
These should **only** appear in the package.mo of Modelica not in any other place as this might confuse Modelica tools. See also section 17.7 of the Modelica Specification 3.1

I know that it might look convenient for library developers to use this version number for tracking there own development of their library. But once the files are integrated into the Modelica Standard Library they **must be removed**. If a library developer still likes to include some note which version actually was integrated into the MSL he should use comments. 

I'm going to remove all those illegal annotations later today and will try to convert them to some sort of comment in the following way: 
 Example taken from `Translational.mo`:
 ```diff
annotation (
-version="1.2.0", versionDate="2009-07-15", 
+// illegal use of top-level annotation removed:
+// version="1.2.0", versionDate="2009-07-15",
Icon(coordinateSystem(preserveAspectRatio=true, extent={{-100,-100},{100,
```

PS: I'm going to do this only for **trunk**, i.e., the to be released MSL 3.1 version, as this was only specified for Modelica 3.1. If a library developer likes to clean up his own code for the maintained releases (especially cleaning up the ugly `uses` annotation) then feel free to do so.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/191

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 15 Jul 2009 08:30 UTC**
Actually `uses` shouldn't appear anywhere in the MSL as it's not depending on any other library (AFAIK). 

Also I thought about a good place to put custom version information of sub-library packages. The best in my opinion is the `revision` part of the documentation rather than hidden in the comments which most normal users never look at (except if the developer actually wants to hide it).

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 15 Jul 2009 11:31 UTC**
In MSL3.1 the Shape object is moved to the new ModelicaServices package. Therefore, there is now a uses annotation for ModelicaServices in MSL3.1 (so the trunk version).

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 15 Jul 2009 11:52 UTC**
OK I'll look out for that one whilst cleaning up.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 15 Jul 2009 12:27 UTC**
I had a look but currently there is no `uses` annotation for ModelicaServices in trunk

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 15 Jul 2009 12:37 UTC**
fixed in 713687cd62601b14d2847288f25c63b9e764e919. The comments containing the version information should hopefully removed later on by the library developers.
